### PR TITLE
EDU-651 expand interactive media chapter interaction

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-651

### CODE REVIEW

**Comment by adrianaluchian** [comment]
Little UI glitch here:

Screen Shot 2022-07-19 at 18 46 13

File: migrations/educandu-2022-07-19-01-rename-question-to-text-in-interactive-media-plugin copy.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      if (docWasUpdated) {
+        updateCount += 1;
+        await this.db.collection(collectionName).replaceOne({ _id: doc._id }, doc);
+      }
+    }
+
+    return updateCount;
+  }
+
+  async collectionDown(collectionName) {
```
Comment (adrianaluchian) [inline migrations/educandu-2022-07-19-01-rename-question-to-text-in-interactive-media-plugin copy.js:32]: Just as a note, a nice little trick for these mirror up/down methods would be to extract a method like `migrate(collectionName, oldProp, newProp)`, and with this we really ensure we didn't forget to update both methods when we find an issue in the `collectionUp` (which happened to me). We did that in `educandu-2022-07-11-01-rename-text-to-copyrightNotice-in-all-plugins`.

File: src/plugins/interactive-media/interactive-media.yml [hunk 2]
```diff
@@ -37,6 +37,9 @@ answers:
 addAnswer:
   en: Add answer
   de: Antwort hinzufügen
+addAnswerInfo:
+  en: "In order to allow the user to examine their knowledge, you can put the text above into a question and add possible answers here:"
```
Comment (adrianaluchian) [inline src/plugins/interactive-media/interactive-media.yml:41]: Maybe "review" would work better than "examine" 🔍

File: src/components/media-player.js [hunk 3]
```diff
@@ -99,6 +141,8 @@ function MediaPlayer({
```
Comment (adrianaluchian) [inline src/components/media-player.js:141]: Seems `trackRef.current` is not initialized on the initial setting of the video source and it throws on line 138:

Screen Shot 2022-07-20 at 09 44 36

I can reproduce this only when setting the video for the first time, on the newly created `interactive-media` section.

**Comment by ahelmberger** [comment]
> Little UI glitch here:
> 
> Screen Shot 2022-07-19 at 18 46 13

Fixed it, they're 100% now

File: migrations/educandu-2022-07-19-01-rename-question-to-text-in-interactive-media-plugin copy.js (hunk 1 above)
Comment (ahelmberger) [reply migrations/educandu-2022-07-19-01-rename-question-to-text-in-interactive-media-plugin copy.js]: I see. I did a similar thing. Maybe we should re-introduce the kind of migration helpers we once had (to process all documents in particular).

File: src/plugins/interactive-media/interactive-media.yml (hunk 2 above)
Comment (ahelmberger) [reply src/plugins/interactive-media/interactive-media.yml]: True, sounds better. Fixed.

File: src/components/media-player.js (hunk 3 above)
Comment (ahelmberger) [reply src/components/media-player.js:145]: Very good catch, fixed it. Thanks!

**Comment by ahelmberger** [comment]
> Looks very good and finally all plugins start having a common style heart_eyes Screen Shot 2022-07-20 at 09 39 02
> 
> And given all the many moving parts and possible scenarios the navigation, playback etc. works nicely +1

Thanks :bow: I also enjoy the UI getting more and more streamlined

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**